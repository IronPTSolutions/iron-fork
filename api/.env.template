
PORT=3000
MONGODB_URI=mongodb://localhost:27017/iron-fork
JWT_SECRET="123"