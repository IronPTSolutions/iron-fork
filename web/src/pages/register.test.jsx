import {
  cleanup,
  fireEvent,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import { beforeEach, describe, expect, test, vi } from "vitest";
import Login from "./login";
import { BrowserRouter } from "react-router-dom";
import { AlertProvider } from "../contexts/alert.context";
import AuthContext from "../contexts/auth.context";
import Register from "./register";
import * as apiService from "../services/api.service";

const RegisterWrapper = () => (
  <BrowserRouter>
    <AlertProvider>
      <Register />
    </AlertProvider>
  </BrowserRouter>
);

describe("<Login/>", () => {
  beforeEach(() => {
    cleanup();

    vi.spyOn(navigator, "geolocation", "get").mockReturnValue({
      getCurrentPosition: vi.fn(),
    });
  });

  test("render successfully", () => {
    // Given

    // When
    render(<RegisterWrapper />);

    // Then
    screen.getByLabelText("Email address");
    screen.getByLabelText("Password");

    expect(document.body).toMatchSnapshot();
  });

  test("happy case", async () => {
    // Given
    vi.spyOn(apiService, "createUser").mockResolvedValue();

    // When
    render(<RegisterWrapper />);
    const email = screen.getByLabelText("Email address");
    const password = screen.getByLabelText("Password");
    const name = screen.getByLabelText("Name");
    const username = screen.getByLabelText("Username");
    const birth = screen.getByLabelText("Birth date");

    fireEvent.change(email, { target: { value: "john@example.com" } });
    fireEvent.change(password, { target: { value: "12345678" } });
    fireEvent.change(name, { target: { value: "John" } });
    fireEvent.change(username, { target: { value: "superjohn" } });
    fireEvent.change(birth, { target: { value: new Date() } });

    fireEvent.submit(screen.getByTestId("register-form"));

    // Then
    await waitFor(() => {
      expect(apiService.createUser).toHaveBeenCalledWith({
        birthDate: "",
        location: {
          type: "Point",
          coordinates: [0, 0],
        },
        email: "john@example.com",
        password: "12345678",
        name: "John",
        username: "superjohn",
      });
    });
  });
});
