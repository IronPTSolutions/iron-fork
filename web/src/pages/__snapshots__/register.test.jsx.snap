// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Login/> > render successfully 1`] = `
<body>
  <div>
    <div>
      <div
        class="container my-4"
      >
        <form
          data-testid="register-form"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="name"
            >
              Name
            </label>
            <input
              class="form-control "
              id="name"
              name="name"
              required=""
              type="text"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
              Email address
            </label>
            <input
              class="form-control "
              id="email"
              name="email"
              required=""
              type="email"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              class="form-control "
              id="password"
              name="password"
              required=""
              type="password"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="username"
            >
              Username
            </label>
            <input
              class="form-control "
              id="username"
              name="username"
              type="text"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="birthDate"
            >
              Birth date
            </label>
            <input
              class="form-control "
              id="birthDate"
              name="birthDate"
              required=""
              type="date"
            />
          </div>
          <button
            class="btn btn-success"
            type="submit"
          >
            Register
          </button>
        </form>
      </div>
    </div>
  </div>
</body>
`;
